csharp/src/Google.Protobuf/ByteString.cs

```diff
@@ -400,7 +400,7 @@ public override int GetHashCode()
         /// </summary>
         /// <param name="other">The <see cref="ByteString"/> to compare this with.</param>
         /// <returns><c>true</c> if <paramref name="other"/> refers to an equal byte string; <c>false</c> otherwise.</returns>
-        public bool Equals(ByteString other)
+        public bool Equals(ByteString? other)
```

Comment:
Interestingly no, IEquatable<T>.Equals() takes T? by definition: https://docs.microsoft.com/en-us/dotnet/api/system.iequatable-1.equals?view=net-6.0#system-iequatable-1-equals(-0)

Reply:
Ah, that's interesting. I think that may have changed since I first played around with NRTs a long time ago. It was always a source of some discomfort for me. Good to see it's sorted!